File: Basic_maths/palindrome.cpp
#include<iostream>
#include<math.h>

using namespace std;


int main(){

    // int n=123;
    int n=120;
    int dup=n;
    int rev=0;

    if (n<=0 || n%10 == 0){
        cout<<"Not Palindrome";
    }
    else{

        while(n !=0){
            
            int last = n%10;
            
            rev=(rev*10)+last;
            
            n/=10;
        }
        
        cout <<"Rev : " << rev <<endl;
        
        if (rev==dup) cout<<"Palindrome";
        else cout<<"Not Palindrome";
    }
  
    return 0;
}

File: Basic_maths/Python/palindrome.py

def rev_number(n):

    rev=0
    while n>0:
        l=n%10

        rev=(rev * 10) + l

        n=n//10
    return rev


n=1231

if n == rev_number(n):
    print("Palindrome...!")
else: 
    print("Not Palindrome...!")